security/ThreatModelling/ThreatModelling.md

```diff
@@ -0,0 +1,93 @@
+> 2.	If you don’t know what to protect, how do you know you’re protecting it?
+> 3.	If you are not protecting it, the dragon wins. 
+
+## What is Threat Modelling
+
+At its most basic level; Threat Modelling is identifying and prioritising potential attacks, threats, and vulnerabilities within your assets. This is done by identifying assets (using methods such as data flow diagrams) to help discuss scenarios, "attack surface" areas, and countermeasures.  
+
+There are different _approaches_ and _methodologies_ to threat modelling that help support different operational functions. "_Approaches_" help support the "_Methodology_" chosen, and neither are used in isolation.  
+
+### Approaches
+
+- Asset based – Best used to describe assets when working with a multi-disciplinary team.
+- Attacker based – Best used when working with experienced security team members.
+- Application based – Best used when working in/with a team made up mostly of developers.
+
+### Methodologies
+
+- __PASTA__ – Process Attack Simulation and Threat Analysis
+- __STRIDE__ – Spoofing, Tampering, Repudiation, Information Disclosure, Denial of Service, Elevation of Privilege
+- __OCTAVE__ – Operationally Critical Threat, Asset and Vulnerability Evaluation
+- __TRIKE__ – Risk based security auditing from top level down.
+- __VAST__ – Visual, Agile, Simple Threat modelling
+
+## How do we perform threat modelling
+
+Threat modelling should be performed: 
+
+ - At the beginning of a project - (to assist with understanding potential attack vectors, and allow for designs to be changed before anything is created),
+ - Whenever there is a change to the _attack surface_ of the application.
+
+Wherever possible, a **_Data Flow Diagram_** can be created to help visualise and define the assets that are apart of the application, so that a conversation can be started about attack vectors of the different areas. The data flow diagram should be a growing diagram that is maintained in tandem with the project.
+
+The tooling of choice to support this is the **_Microsoft Threat Modelling Tool_**. Utilising the data flow diagram within the tooling, it can analyse and support the STRIDE method of giving potential attack vectors to your application. This tooling should be a compliment for the _conversation_ however, rather than a replacement for it. 
+
+To start the conversation for a Threat Modelling session, the following will need to be understood so that the session can focus on what is required from the threat modelling:
+
+- __Scope__ – What is in scope for this session?
+- __Stakeholders__ – Who has input into this system design and needs to be involved in this modelling.
+- __Risk management__ – Who will support you in assessing and devising a plan for any risks that are found.
+
+Depending on the stage of the project, facilitation can be handled in two ways:
+
+- Early project – Facilitated by Security personnel (Principals, Lead Tech, ITSO)
```

Comment:
Is STRIDE done at the start of project? The gov link suggested it would be run by a PM or the secuirty org, with the relevent expertise from devs, devOps, hardware admin, net admin.

Reply:
I have just been conversing with Dave Tilsley on this and at the start of the project (before any code is written) the ideal would be to have a data flow diagram and relevant people present (customers, user researchers/designers, security, devs, operations) so that a fluent discussion can be had about the initial design. Then we can move onto a more technical TM session as described above before being a little more hands off. STRIDE is how we categorise the things we find so it will be done all the way through, its just who is involved at what point may change.